### ISSUE TITLE
Add battery_quantity attribute to battery_type entity

### ISSUE THREAD

**Issue by nechry** [andrew-codechimp/HA-Battery-Notes#375]
### Checklist

- [X] I have filled out the template to the best of my ability.
- [X] This only contains 1 feature request (if you have multiple feature requests, open one feature request for each feature request).
- [X] This issue is not a duplicate feature request of [previous feature requests](https://github.com/andrew-codechimp/ha-battery-notes/issues?q=is%3Aopen+is%3Aissue+label%3Aenhancement).

### Is your feature request related to a problem? Please describe.

first of all thank you for this integration. I previously built my battery information base using customization (customize.yaml) like this:
```yaml
sensor.first_floor_landing_motion_sensor_battery_level:
  battery_type: CR123A
sensor.first_floor_landing_smoke_sensor_battery_level:
  battery_type: CR123A
sensor.first_floor_landing_lcd_battery:
  battery_type: LR03/AAA
  quantity: 2
sensor.aqara_opple_switch_2x2_battery:
  battery_type: CR2032
sensor.anemometer_battery:
  battery_type: LR06/AA
sensor.attic_staircase_motion_sensor_battery:
  battery_type: CR2450
```

This way I added the battery type and quantity to the device's existing battery. You can find more details on my github https://github.com/nechry/ha-battery-inventory

The goal is to reproduce a list of batteries by battery type with the total number of batteries by type and % charge remaining. This is very useful for planning battery ordering.

### Describe the solution you'd like

If it would be possible to add the quantity to the battery_type information as an attribute it would be very useful for producing battery reports

### Describe alternatives you've considered

it is possible to parse the text sensor with a regex to cut the quantity of the type but I think it would still be cumbersome

### Additional context

nothing to add at first but remains available to discuss it

**Comment by andrew-codechimp** [comment]
Nice idea, I can add some attributes to split the quantity and type, ironically I'll have to write a regex as well as I have it as one combined string when it is assigned to the device.

**Comment by nechry** [comment]
great thank for your positive feedback, next step will be to mix information of the battery level with type to reproduce my previous battery remplacement report.

**Comment by andrew-codechimp** [comment]
I have a rough version working with a bit of regex

![CleanShot 2024-01-06 at 11 47 44@2x](https://github.com/andrew-codechimp/HA-Battery-Notes/assets/1849731/6000f648-46e9-4d56-ba41-2d9582953559)

And yes I always test with Sun & Forecast because they are always in a fresh HA install 😄

**Comment by andrew-codechimp** [comment]
This is now available in version 1.4.0 which is in pre-release

**Comment by nechry** [comment]
thank-you I will try it soon
